### DISCUSSION TITLE
Part of a sprite draw

### DISCUSSION THREAD

**Discussion by Tinkerpete** [Bodmer/TFT_eSPI#864] [Q&A - Sprites] [answered]
What is the best way to draw only a part of a sprite to a certain position on the TFT display?

**Comment by Bodmer** [comment]
Use [this](https://github.com/Bodmer/TFT_eSPI/blob/master/Extensions/Sprite.h#L125) function.

**Comment by Tinkerpete** [comment]
I didn't know that 'pushSprite' can also be called with a window.
Thanks!

For others:

// Push a windowed area of the sprite to the TFT at tx, ty
pushSprite(int32_t tx, int32_t ty, int32_t sx, int32_t sy, int32_t sw, int32_t sh);

**Comment by Bodmer** [comment, marked as answer]
Example use:
```
tft.pushSprite(tft_x, tft_y, sprite_x, sprite_y, window_width, window_height);
```

**Comment by ha-ina** [comment]
Wouldn't it be great if the function with a window in the destination was also possible for pushToSprite?
This would be very useful for one of my problems.
Thank you.

**Comment by Bodmer** [comment]
Use the setViewport function to set a viewport window in the destination sprite, then pushToSprite with the required coordinate offsets.
